import-service/src/handlers/importProductsFile/importProductsFile.ts

```diff
@@ -0,0 +1,50 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import AWS from 'aws-sdk';
+import Joi from 'joi';
+import { defaultCors } from 'lib/constants/cors';
+
+export const importProductsFile: APIGatewayProxyHandler = async (event) => {
+  try {
+    console.log('queryStringParameters', event.queryStringParameters);
+
+    const name = event.queryStringParameters?.name;
+
+    const { error } = Joi
+      .string()
+      .pattern(/^[\w,\s-]+\.csv$/, { name: 'csv' })
+      .required()
+      .validate(name);
+
+    if (error) {
+      return {
+        statusCode: 400,
+        headers: defaultCors,
+        body: error?.message,
+      };
+    }
+
+    const s3 = new AWS.S3();
+
+    const params = {
+      Bucket: process.env.BUCKET_NAME,
+      Key: `uploaded/${name}`,
+      ContentType: 'text/csv',
+      ACL: 'public-read'
+    };
+
+    const signedUrl = await s3.getSignedUrlPromise('putObject', params);
+
+    return {
+      statusCode: 202,
```

Comment:
Just for interest. I thought we need to simply return 200 as we actually did a request and got the result (signed url).
And 202, from my understanding, better suites for some async operations, when we cannot receive a result immediately
But I'm not 100% sure =))

Reply:
Yes, maybe you are right =) But I thought that we need to return 202 because of we tried to create a file on BE side, and this name has been accepted, and after that we can upload this file =)